pyramid_oereb/contrib/stats/decorators.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import json
+
+LOG = logging.getLogger('JSON')
+
+
+def log_response(wrapped):
+    def wrapper(context, request):
+        response = wrapped(context, request)
+        ret_dict = _serialize_response(response)
+        ret_dict.update(_serialize_request(request))
+        LOG.info(json.dumps(ret_dict))
+        return response
+    return wrapper
+
+
+def _serialize_response(response):
+    x = {}
+    x['status_code'] = response.status_code
+    x['headers'] = dict(response.headers)
+    x['extras'] = response.extras if hasattr(response, 'extras') else None
+    return {'response': x}
+
+
+def _serialize_request(request):
+    x = {}
+    x['headers'] = dict(request.headers)
+    x['traversed'] = str(request.traversed)
+    x['parameters'] = dict(request.GET)
+    x['path'] = str(request.path)
+    x['view_name'] = str(request.view_name)
+    return {'request': x}
+
+
+class OerebStats(dict):
+    """
+    class OerebStats(dict)
+    this class is used to provide a serializable object that can be
+    used to provide insight of application logic in the logs.
+    """
+    def __init__(self,
+                 service=None,
+                 output_format=None,
+                 params=None):
+        super(OerebStats, self).__init__(service=service,
+                                         output_format=output_format,
+                                         params=params)
+        self.itemlist = super(OerebStats, self).keys()
+
+    def __setitem__(self, key, value):
+        # TODO: what should happen to the order if
```

Comment:
What about this TODO? Do we need to decide something?

Reply:
Actually, we just have to let it behave like a normal dictionary. Solved [here](https://github.com/danduk82/pyramid_oereb/pull/2/commits/5b8c26c9b5c82d002d0b3903e0a4b2e479002b8f)